src/Parser/DemonologyWarlock/Modules/Talents/Implosion.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import SPELLS from 'common/SPELLS';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber, formatPercentage } from 'common/format';
+
+class Implosion extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  impsKilled = 0;
+  damage = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.IMPLOSION_TALENT.id);
+  }
+
+  on_byPlayer_instakill(event) {
+    if (event.ability.guid !== SPELLS.IMPLOSION_DAMAGE.id) {
+      return;
+    }
+    this.impsKilled += 1;
+  }
+
+  on_byPlayer_damage(event) {
+    if (event.ability.guid !== SPELLS.IMPLOSION_DAMAGE.id) {
+      return;
+    }
+    this.damage += event.amount + (event.absorbed || 0);
+  }
+
+  statistic() {
+    return (
+      <StatisticBox
+        icon={<SpellIcon id={SPELLS.IMPLOSION_TALENT.id} />}
+        value={`${formatNumber(this.damage / this.owner.fightDuration * 1000)} DPS`}
+        label="Implosion damage"
+        tooltip={`Your Implosion did ${formatNumber(this.damage)} total damage (${formatPercentage(this.owner.getPercentageOfTotalDamageDone(this.damage))} %) and killed ${this.impsKilled} Wild Imps.`}
```

Comment:
Is the damage lost from killing your Imps early something that needs to be considered?

Reply:
That is most likely correct, I think I've read that it's recommended to let them live for about 8 of their 12 second lifespan, then summon a new set and right before they spawn (the meteor that spawns them has about 0,7s delay) bomb them, so the new set survives. I'll keep this as an initial basic implementation because Implosion is mostly M+ talent, and not very used in raids (even on add heavy fights like Demonic Inquisition) and I'll try to look into that later.

Reply:
This might be a good 'recommendation' item to add in the future.